CMakeLists.txt

```diff
@@ -231,13 +231,13 @@ endif ()
 
 include(StaticAnalyzers)
 if (OPENASSETIO_ENABLE_CLANG_FORMAT)
-    enable_clang_format()
+    SET_CLANG_FORMAT()
```

Comment:
[question] Not really loving the uppercase here, looks too much like a variable, and pretty much all functions I've seen in other projects are lowercased. How easy would it be to allow `lower_snake_case` for functions/macros? (is there a distinction between functions and macros?).

Reply:
Fair enough, I was just sticking to the default style, but will change macros to have similar requirements to functions.

That being the case, there's no need in my mind to distinguish these symbols from the variables, so will change the names back to ENABLE_

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/811/commits/99f68bddec6ad40fa6d7ec416f40c6fcc44b5e8e

Reply:
Despite the discussion at standup, I'd rather go with your intuition that most macros seem to follow function styling (lowercase) from the cmake you've seen, rather than the cmake-lint default, if you're agreeable.